### PR TITLE
feat: added input and select components

### PR DESCRIPTION
**Pull request by alexzavgorodnii**
image

### CODE REVIEW

File: src/shared/components/Input/component.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import {
+  InputRootContainer,
+  InputContainer,
+  InputWrapper,
+  InputValidationMessageWrapper,
+  InputGroupButton,
+} from './styles';
```
Comment (braindrain9) [inline src/shared/components/Input/component.tsx:8]: please make them one-liner to save space, like
```js
import { InputRootContainer, InputContainer, InputWrapper, InputValidationMessageWrapper, InputGroupButton } 
  from './styles';

// or if it's not enough space

import { InputRootContainer, InputContainer, InputWrapper, InputValidationMessageWrapper, 
  InputGroupButton } from './styles';
```

File: src/shared/components/Input/component.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import { SendIcon } from '../../icons';
```
Comment (braindrain9) [inline src/shared/components/Input/component.tsx:9]:
```js
import { SendIcon } from 'shared/icons';
```
also, absolute import should go higher

File: src/shared/components/Select/component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { SelectedWrapper, SelectWrapper, SelectItemsList, SelectItemsListItem } from './styles';
+import { DropdownIcon } from '../../icons';
```
Comment (braindrain9) [inline src/shared/components/Select/component.tsx:3]: 'shared/icons';

File: src/shared/components/Select/component.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  onSelect: (value: string) => void;
+};
+
+export type SelectItem = {
+  value: string;
+  name: string;
+};
+
+const Select: React.FC<SelectProps> = ({ placeholder, options, onSelect }: SelectProps) => {
+  const [openedSelect, setOpenedSelect] = useState<boolean>(false);
```
Comment (braindrain9) [inline src/shared/components/Select/component.tsx:17]: let's call `openedSelect ` as `isOpenedSelect `

File: src/shared/components/Select/component.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import { SelectedWrapper, SelectWrapper, SelectItemsList, SelectItemsListItem } from './styles';
+import { DropdownIcon } from '../../icons';
+
+export type SelectProps = {
```
Comment (braindrain9) [inline src/shared/components/Select/component.tsx:5]: let's put these type into `shared/types`; and import from there.

File: src/shared/components/Select/styles.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
+import styled from 'styled-components';
+import Colors from 'shared/colors';
+
+type SelectProps = {
+  openedSelect: boolean;
```
Comment (braindrain9) [inline src/shared/components/Select/styles.ts:5]: if it's boolean let's better call it `isOpenedSelect `

File: src/shared/components/Select/styles.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  ${({ openedSelect }: SelectProps): string => (openedSelect ? dropdownTransformRotate : '')};
+`;
+export const SelectedWrapper = styled.span`
+  ${selectFontStyle};
+  color: ${Colors.BLACK};
+`;
+
+export const SelectItemsList = styled.div`
+  width: 100%;
+
```
Comment (braindrain9) [inline src/shared/components/Select/styles.ts:41]: please avoid empty rows

File: src/shared/icons/index.ts [hunk 8]
```diff
@@ -6,5 +6,18 @@ import LogoWhiteIcon from './LogoWhite';
 import LinkedinIcon from './Linkedin';
 import TwitterIcon from './Twitter';
 import FacebookIcon from './Facebook';
+import SendIcon from './Send';
+import DropdownIcon from './Dropdown';
 
-export { SearchIcon, CloseIcon, LogoBlackIcon, BoxIcon, LogoWhiteIcon, LinkedinIcon, TwitterIcon, FacebookIcon };
+export {
```
Comment (braindrain9) [inline src/shared/icons/index.ts:12]: let's make them in alphabetical order

File: src/shared/components/Input/component.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+export type InputProps = React.HTMLAttributes<HTMLInputElement> & {
+  inputType: string;
+  validationState: boolean;
+  validationMessage: string;
+  withSend?: boolean;
```
Comment (braindrain9) [inline src/shared/components/Input/component.tsx:15]: I am not sure about `withSend` naming, it looks a bit confusing for me. 

Let's call it like `includeSendButton` to be more clear.

File: src/shared/colors.ts [hunk 10]
```diff
@@ -16,6 +16,7 @@ enum Colors {
   BLACK = '#000000',
   WHITE = '#FFFFFF',
   DARK_BACKGROUND = '#181818',
+  VALIDATION_ERROR = '#F14517',
```
Comment (braindrain9) [inline src/shared/colors.ts:19]: Please name it according to Color naming convention, like RED or RED_BRIGHT

File: src/shared/components/Input/component.tsx (hunk 1 above)
Comment (alexzavgorodnii) [reply src/shared/components/Input/component.tsx:8]: It's prettier settings.  When I try to change the format it's automatically changed to this.

File: src/shared/colors.ts (hunk 10 above)
Comment (alexzavgorodnii) [reply src/shared/colors.ts]: done

File: src/shared/components/Input/component.tsx (hunk 9 above)
Comment (alexzavgorodnii) [reply src/shared/components/Input/component.tsx]: done

File: src/shared/icons/index.ts (hunk 8 above)
Comment (alexzavgorodnii) [reply src/shared/icons/index.ts:12]: done

File: src/shared/components/Select/styles.ts (hunk 7 above)
Comment (alexzavgorodnii) [reply src/shared/components/Select/styles.ts]: done

File: src/shared/components/Select/styles.ts (hunk 6 above)
Comment (alexzavgorodnii) [reply src/shared/components/Select/styles.ts]: done

File: src/shared/components/Select/component.tsx (hunk 5 above)
Comment (alexzavgorodnii) [reply src/shared/components/Select/component.tsx]: done

File: src/shared/components/Select/component.tsx (hunk 4 above)
Comment (alexzavgorodnii) [reply src/shared/components/Select/component.tsx]: done

File: src/shared/components/Select/component.tsx (hunk 3 above)
Comment (alexzavgorodnii) [reply src/shared/components/Select/component.tsx]: done

File: src/shared/components/Input/component.tsx (hunk 2 above)
Comment (alexzavgorodnii) [reply src/shared/components/Input/component.tsx]: done

File: src/shared/components/Input/component.tsx (hunk 1 above)
Comment (braindrain9) [reply src/shared/components/Input/component.tsx]: oh, I see, I think we should fix this in future.

**Comment by braindrain9** [comment]
please resolve merge conflicts and it's ready to go 💯

File: src/shared/components/Input/styles.ts [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  validationState?: boolean;
+  validationMessage?: string;
+};
+
+export const InputRootContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: flex-start;
+  justify-content: flex-start;
+  width: 100%;
```
Comment (braindrain9) [inline src/shared/components/Input/styles.ts:16]:
```css
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  width: 100%;
```
could be moved to a shared variable and used in InputRootContainer and InputContainer

Comment (alexzavgorodnii) [reply src/shared/components/Input/styles.ts]: Done

**Pull request merged**